utility/environment_check.py

```diff
@@ -0,0 +1,127 @@
+"""
```

Comment:
Instead of utility function how about handling it in OCS.create() and OCS.delete() itself. Lets say we add a decorator @register_resource for create() and delete() functions. Every create will register (resource_kind, name, namespace) in a global list and every delete() will remove that element from global list. 

1. If we try to delete the resource which we haven't created, that element will not be in list so raise exception
2. If we created element but forgot to delete, during finalize just scan that list and raise exception.

Reply:
To make it work on every create() delete(), you will need to maintain a map of all resources creation and deletion so you will know what is created as part of the resource.
For example, when I create a cephfilesystem, also 2 mds pods are being created so how can I count it?
I think the approach you are suggesting is much more complicated and doesn't add too much value.

Reply:
> To make it work on every create() delete(), you will need to maintain a map of all resources creation and deletion so you will know what is created as part of the resource.
> For example, when I create a cephfilesystem, also 2 mds pods are being created so how can I count it?
> I think the approach you are suggesting is much more complicated and doesn't add too much value.

I agree. I would rather keep this simple at first. We can revisit the labeling idea (as noted in different comment) later, if we found it necessary, but right now I don't have enough experience to guess if that will be useful or not.

Reply:
> To make it work on every create() delete(), you will need to maintain a map of all resources creation and deletion so you will know what is created as part of the resource.
> For example, when I create a cephfilesystem, also 2 mds pods are being created so how can I count it?
> I think the approach you are suggesting is much more complicated and doesn't add too much value.

In this specific case it could still work, if you are expecting the test case to create and delete filesystem, we will be tracking the resource like this [(myfs, openshift-storage,CephFilesystem)]. During cleanup just delete myfs so any resource created as part of this fs will be wiped out.

If you want to create fs here and delete it elsewhere then this method won't work.